src/bindings/gaiabuild/view.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+import { Env } from '../../lib/env';
+import { qps } from '../../lib/pseudo';
+import { getResourceLinks } from '../../bindings/html/head';
+import { translateFragment } from '../../bindings/html/dom';
+import { getDirection } from '../../bindings/html/langs';
+import { serializeContext } from '../../bindings/gaiabuild/serialize';
+
+export class View {
+  constructor(htmloptimizer, fetch) {
+    this.env = new Env(
+      htmloptimizer.config.GAIA_DEFAULT_LOCALE, fetch);
+    this.htmloptimizer = htmloptimizer;
+    this.doc = htmloptimizer.document;
+    this.ctx = this.env.createContext(getResourceLinks(this.doc.head));
+
+    // add the url of the currently processed webapp to all errors
+    this.env.addEventListener('*', amendError.bind(this));
+
+    this.stopBuildError = null;
+    const log = logError.bind(this);
+    const stop = stopBuild.bind(this);
+
+    // stop the build if these errors happen for en-US
+    // XXX tv_apps break the build https://bugzil.la/1179833
+    // this.env.addEventListener('fetcherror', stop);
+    // XXX parse errors don't have the lang property needed by stopBuild
```

Comment:
Then I'd argue that we should intercept the error, add a lang and push it forward instead of transforming parser API to be aware of the lang it's parsing. Parser should be dumb imho.